cli/src/main/kotlin/commands/ClearlyDefinedUploadCommand.kt

```diff
@@ -0,0 +1,135 @@
+
+    @Parameter(
+        description = "The ClearlyDefined server to upload to.",
+        names = ["--server", "-s"],
+        order = PARAMETER_ORDER_OPTIONAL
+    )
+    private var server = Server.DEVELOPMENT
+
+    private fun PackageCuration.toContributionPatch(): ContributionPatch {
+        val info = ContributionInfo(
+            // The exact values to use here are unclear; use what is mostly used at
+            // https://github.com/clearlydefined/curated-data/pulls.
+            type = "Import",
+            summary = "Curation for package with ${id.toClearlyDefinedCoordinates()}.",
+            details = "Imported from curation data of the " +
+                    "[OSS Review Toolkit](https://github.com/heremaps/oss-review-toolkit) via the " +
+                    "[clearly-defined](https://github.com/heremaps/oss-review-toolkit/tree/master/clearly-defined) " +
+                    "module.",
+            resolution = data.comment ?: "Unkown, original data contains no comment.",
+            removedDefinitions = false
+        )
+
+        val licenseExpression = data.concludedLicense?.toString() ?: data.declaredLicenses?.joinToString(" AND ")
+
+        val curation = Curation(
+            described = Described(
+                projectWebsite = data.homepageUrl?.let { URL(it) },
+                sourceLocation = toClearlyDefinedSourceLocation(id, data.vcs, data.sourceArtifact)
+            ),
+            licensed = licenseExpression?.let { Licensed(it) }
+        )
+
+        val patch = Patch(
+            coordinates = id.toClearlyDefinedCoordinates(),
+            revisions = mapOf(id.version to curation)
+        )
+
+        return ContributionPatch(info, listOf(patch))
+    }
+
+    override fun runCommand(jc: JCommander, config: OrtConfiguration): Int {
+        val absoluteInputFile = inputFile.expandTilde().normalize()
+
+        require(absoluteInputFile.isFile) {
+            "The file '$absoluteInputFile' could not be found."
+        }
+
+        val curations = absoluteInputFile.readValue<List<PackageCuration>>()
+
+        val service = ClearlyDefinedService.create(server)
+
+        curations.forEachIndexed { index, curation ->
+            val patchCall = service.putCuration(curation.toContributionPatch())
+            val response = patchCall.execute()
+            val responseCode = response.code()
+
+            print("Curation $index of ${curations.size} for package '${curation.id.toCoordinates()}' ")
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                response.body()?.let { responseBody ->
+                    val summary = jsonMapper.treeToValue(responseBody, ContributionSummary::class.java)
+                    println("was successfully uploaded:\n${summary.url}")
+                } ?: log.warn { "The REST API call succeeded but no response body was returned." }
+            } else {
+                println("failed to be uploaded (response code $responseCode).")
```

Comment:
If there is also a response message available it would be nice to log it as well.

Reply:
The response message in case of an error is (only) contained in the error body which is logged a few lines below.